### PR TITLE
Gem plugin support

### PR DESCRIPTION
**Pull request by matthutchinson**
This PR introduces a new PluginManager class that can load plugins from external gems (and include them alongside the existing plugins already present).  

For reviewing - much of this takes place in the new `Lolcommits::PluginManager`, `Lolcommits::GemPlugin`, and `Lolcommits::Plugin::Base` classes. 

The approach is very similar to how the Pry gem manages plugins. Plugins can declare their compatibility with the main lolcommits gem version in their Gemspec. This should (hopefully) prevent gem dependecy issues from spiralling out of control over time.

Lolcommit gems must be prefixed with the name `lolcommits-plugin-{whatever}`. A new showcase example gem plugin is ready [here](https://github.com/lolcommits/lolcommits-plugin-sample) and will become releaseable after this PR has been merged (and a new lolcommits gem released). This gem is a poster-boy for good plugin development.  

With this PR released, I can begin to move all other plugins out to gems one by one. Following a similar pattern for each.  After that I'll update [this section](https://github.com/lolcommits/lolcommits-plugin-sample#developing-your-own-plugin) of the example gem README with more details.

Other things included in this PR:

* All plugins are now in a new namespace `Lolcommits::Plugin::PluginName`
* Added `lib/lolcommits/test_helpers/` that can be required in plugins for easier test setup
* A readable `DEFAULT_FONT_PATH` is no longer the concern of the main gem
* `valid_configration?` is now called for all plugins by the runner in pre/post capture hooks
* License info added to README

Future considerations:

* Make use of the `lib/lolcommits/test_helpers/` in the main gem, switch to MiniTest?
* To avoid duplicated plugins, have the plugin name be auto-extracted from the Gem name itself, so this concern is handled by RubyGems.
* After extracting all plugins remove any code that loads plugins from inside the main gem.
* Update the README to link to the example gem, explaining how to develop plugins.

I started a lot of this work around Christmas period, but I'm only finishing now!

### CODE REVIEW

**Comment by matthutchinson** [comment]
Closes #99

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ ensure Ruby style guidelines. If you want to run tests before conforming to this
 to verify functionality, just run `rake test` and `rake features` manually.
 
 With a passing test suite, commit your changes, push and submit a new [Pull
-Request](https://github.com/mroth/lolcomits/compare/):
+Request](https://github.com/mroth/lolcommits/compare/):
```
Comment (mroth) [inline CONTRIBUTING.md]: lol whoops

File: README.md [hunk 2]
```diff
@@ -209,3 +209,10 @@ issue](https://github.com/mroth/lolcommits/issues) (and please take a little
 time to check if we haven't [already
 addressed](https://github.com/mroth/lolcommits/issues?q=is%3Aissue+is%3Aclosed)
 it).
+
+
+## License
+
+The gem is available as open source under the terms of
```
Comment (mroth) [inline README.md]: Nit: `s/gem/program/` to leave a Rubyism out of the legal info.

Comment (matthutchinson) [inline README.md]: 👍

File: CONTRIBUTING.md (hunk 1 above)
Comment (matthutchinson) [inline CONTRIBUTING.md]: 😄

**Comment by matthutchinson** [comment]
Yep, for releasing i think this warrants moving from `0.9.2` to `0.10.0` - then after all plugins have been moved, we could consider finally releasing a `v1.0`

**Comment by matthutchinson** [comment]
Thanks for the review, will merge and prep a release later today.

**Comment by matthutchinson** [comment]
🍈 just testing the v0.9.3.pre1 (pre-release) and noticed a require issue with the PluginManager, will attempt to debug/fix asap.

**Comment by matthutchinson** [comment]
Fixed now, and sample plugin [gem released](https://rubygems.org/gems/lolcommits-plugin-sample) - a few more checks on other platforms and I can release 0.9.3

**Comment by matthutchinson** [comment]
OK, so before I gem release this version. I've just thought of a potential security issue.

This change effectively means our lolcommits gem is now open unverified code execution. Consider this scenario.

* Malicious dev creates a lolcommits plugin gem with `enabled?` and `configured?` subclassed and returning true
* Malicious dev tricks someone into installing this, either directly or by masking the gem (as a dependency on another gem called 'free-cash-for-fun' or something)
* Unsuspecting dev installs lolcommits, behind the scenes, this plugin will always run (silently) on the post capture hook
* The code in the unauthorised plugin runs and can do any number of terrible things e.g.;
    * Look at `self.runner.vcs_info` extract git repo info and fork a process to remotely copy their repo
    * Copy lolcommit snapshots, or inject a cron to use the lolcommit subsystem itself to periodically take snapshots and send to a remote server
    * system("rm -rf /") 💀 
    * Rick roll you everytime you commit `system("open http://youtube.com/...")`

@mroth - do you think I'm being over cautious?

With this in mind, I think we should change the PluginManager to validate plugins against a list of qualified/verified gem names maintained in this gem (rubygems ensures the gem names are unique).

It would make plugin development a 2 step process (create and publish gem, then small PR to lolcommits) -- but would protect us from the above 🛡 . What do you think ?

Ps. if we go with this approach, I will also change the PluginManager to derive gem plugin names from the gem name itself.

**Comment by matthutchinson** [comment]
I'll prep these suggested changes in a PR and we can review when your'e back on the 17th. Also I can try to migrate a bunch of the existing plugins to gems.

**Comment by matthutchinson** [comment]
Rethinking this again, since Ruby is inherently open, it is kind of pointless to put any restrictions in place. Any gem could easily override the relevant methods to circumvent things. That said, I'm going to make some more improvements with gem/plugin naming in another PR.

**Pull request merged**